File: network/src/main/java/com/network/base/BaseResponseModel.java
package com.network.base;

import androidx.annotation.NonNull;

import com.google.gson.Gson;
import com.google.gson.JsonObject;
import com.google.gson.annotations.SerializedName;


public class BaseResponseModel<B> extends ResponseModel{

    @SerializedName("data") // @Json(name="data")
    private JsonObject data;

    public B getResponseModel(Class<B> aClass) {
        return prepareModel(data, aClass);
    }

    private <T> T prepareModel(@NonNull JsonObject aString, Class<T> aClass) {
        return new Gson().fromJson(aString, aClass);
    }
}

File: network/src/main/java/com/network/base/BaseResponseModel.kt
package com.network.base
import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
class BaseResponseModel<T> {
    @Json(name = "status")
    var status = 0

    @Json(name = "message")
    var message: String? = null

    @Json(name = "statusCode")
    var statusCode = 0

    @Json(name = "data")
    var data: T? = null
}